source/distributed/tria.cc

```diff
@@ -3582,21 +3686,35 @@ namespace parallel
                 balance_type(P8EST_CONNECT_FULL)),
                /*init_callback=*/NULL);
 
+      // before repartitioning get cell weights for a weighted repartitioning.
+      // TODO: move this call into the if below, only call when slots are connected
+      std::vector<unsigned int> cell_weights;
+      get_cell_weights(cell_weights);
```

Comment:
In recent times, we've gone more towards making functions like this actually return the object, rather than taking the return object as last argument. Can you change this so that `get_cell_weights` returns the vector? This also allows you to assign it to a `const` vector.